exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntime.java

```diff
@@ -206,6 +206,23 @@ public void initiateAddingService(Fork fork, ServiceInstanceSpec instanceSpec,
     }
   }
 
+  /**
+   * Initiates resuming of previously stopped service instance.
+   *
+   * @param instanceSpec a service instance specification; must reference a deployed artifact
+   * @param configuration service instance configuration parameters as a serialized protobuf
+   *     message
+   * @throws IllegalArgumentException if the given service instance is active
+   */
+  public void initializeResumingService(ServiceInstanceSpec instanceSpec,
+      byte[] configuration) {
+    synchronized (lock) {
+      checkStoppedService(instanceSpec.getId());
+      ServiceWrapper service = createServiceInstance(instanceSpec);
+      service.resume(new ServiceConfiguration(configuration));
+    }
```

Comment:
Currently this class is responsible to log errors, though there is ECR-3813 to shift that.